bundle/src/main/java/com/adobe/acs/commons/properties/filter/TemplatedFilter.java

```diff
@@ -0,0 +1,268 @@
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.metatype.annotations.AttributeDefinition;
+import org.osgi.service.metatype.annotations.Designate;
+import org.osgi.service.metatype.annotations.ObjectClassDefinition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+@Component(
+        service = Filter.class,
+        property = {
+                Constants.SERVICE_RANKING + ":Integer=" + Integer.MIN_VALUE,
+                EngineConstants.SLING_FILTER_SCOPE + "=" + EngineConstants.FILTER_SCOPE_REQUEST,
+                EngineConstants.SLING_FILTER_PATTERN + "=/content/.*"
+        },
+        configurationPolicy = ConfigurationPolicy.REQUIRE
+)
+@Designate(ocd = com.adobe.acs.commons.properties.filter.TemplatedFilter.Config.class)
+public class TemplatedFilter implements Filter {
+
+    private static final Logger log = LoggerFactory.getLogger(TemplatedFilter.class);
+
+    @Reference
+    private PropertyAggregatorService propertyAggregatorService;
+
+    private Map<String, Object> properties;
+    private List<Pattern> includePatterns;
+    private List<Pattern> excludePatterns;
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        // do nothing
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
+        RequestPathInfo currentPathInfo = ((SlingHttpServletRequest) servletRequest).getRequestPathInfo();
+        if (StringUtils.equals(currentPathInfo.getExtension(), "json") && shouldProcess(((SlingHttpServletRequest) servletRequest).getPathInfo())) {
+            CapturingResponseWrapper capturingResponseWrapper = new CapturingResponseWrapper((SlingHttpServletResponse) servletResponse);
+            filterChain.doFilter(servletRequest, capturingResponseWrapper);
+
+            SlingHttpServletRequest slingHttpServletRequest = (SlingHttpServletRequest) servletRequest;
+
+            String currentResponse = capturingResponseWrapper.getCaptureAsString();
+            String toReturn = currentResponse;
+            try {
+                properties = propertyAggregatorService.getProperties(slingHttpServletRequest);
```

Comment:
Would this be more clearly named `variableReplacementMap` or `variableReplacements`?

Reply:
Good point - updated in this file and the Transformer